store/postgres/migrations/000006_create_discussions_table.up.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE discussions (
+  id serial PRIMARY KEY,
+  title text NOT NULL,
+  body text NOT NULL,
+  state integer NOT NULL DEFAULT 0,
+  type integer NOT NULL DEFAULT 0,
+  owner uuid NOT NULL REFERENCES users(id) ON DELETE CASCADE,
```

Comment:
hmm im thinking to restrict it instead of cascade, missing discussion can be confusing later on.
just a thought, wdyt?

Reply:
if we restrict it, we can't remove user id in the users table. If that's the reason, we can remove the foreign key of this. So, discussion is immutable.

Reply:
hmm yes we won't be to remove a user if he/she has created a discussion. I'm still comparing the downside is better if we cascadingly delete a discussion which could have evolved and store an important information.

or we can use `set null`? wdyt?

Reply:
Yes discussion should not be deleted.

Hmm I don't think `null` would help. I think ownership information should always be there regardless user exists or not in user table. It is okay I guess to not have FK. The worst case would be we have empty `user.User` when doing left join the discussion with user table.